ramalama/model.py

```diff
@@ -633,3 +636,29 @@ def distinfo_volume():
         return ""
 
     return f"-v{path}:/usr/share/ramalama/{dist_info}:ro"
+
+
+def is_port_in_use(port: int) -> bool:
+    with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+        return s.connect_ex(('localhost', port)) == 0
+
+
+def validate_port(port:str, debug:bool) -> str:
+    if port != DEFAULT_PORT:
+        # user custom port, don't override the choice
+        return port
+    port = int(port)
+    found = False
+    # loop until we find an open port
+    while port <= MAX_PORT:
+        if not is_port_in_use(port):
+            found = True
+            break
```

Comment:
Rather than validating port, can we just try and bind to it. And increment if it fails? This solution is less prone to race conditions. I'd also only attempt to bind to 100 ports (less than max port), if they all fail, likely something is drastically wrong, failing ~65k times  could take a while.

It's also worse for performance to check before attempting to connect, we should assume we can connect and try the next port if we fail.

Reply:
makes sense to limit the number of attempts, thanks
> It's also worse for performance to do a test connect before attempting to connect, we should assume we can connect and try the next port if we fail.

Can you expand this? afaik the real connection is done inside the runtime, so are you saying to try to parse the error there and try again to launch the command?